src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -333,10 +333,7 @@ public async Task AfterLoadInitialConfigurationAsync()
         // Ensure the project is not considered loaded until our first publication.
         Task result = _tasksService.PrioritizedProjectLoadedInHostAsync(async () =>
         {
-            using (_joinableTaskCollection.Join())
-            {
-                await WhenInitialized(_tasksService.UnloadCancellationToken);
-            }
+            await WhenInitialized(_tasksService.UnloadCancellationToken);
```

Comment:
we can remove both async/await here, and just return the result of WhenInitialized.

Reply:
I kept this intentionally. Without it, my concern was that a hang related to this code path would be harder to identify as the call to `WhenInitialized` wouldn't be associated at all with the `ProjectAutoLoad` method. Does that make sense? The overhead in this instance is minimal (one additional `Task` per unconfigured project during solution load).